Fluid/TemplateContext.cs

```diff
@@ -150,6 +150,15 @@ public void IncrementSteps()
         /// </summary>
         public Func<string, string, ValueTask<string>> Captured { get; set; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="StringComparer"/> used for matching keys of scope values.
+        /// </summary>
+        public StringComparer StringComparer
```

Comment:
I think you can use the one that is declared in the `TemplateOptions`

https://github.com/sebastienros/fluid/blob/3f1750f690b5b2e02cd13e6554e0df832dbd8385/Fluid/TemplateContext.cs#L85